core/src/main/java/org/apache/iceberg/SnapshotParser.java

```diff
@@ -155,7 +213,11 @@ static Snapshot fromJson(JsonNode node) {
           operation,
           summary,
           schemaId,
-          manifestList);
+          manifestList,
+          manifestListSize,
+          manifestListKeyMetadata,
+          wrappedManifestListKeyMetadata,
+          wrappedKeyEncryptionKey);
```

Comment:
This should create the `ManifestListFile` and pass it in rather than relying on `Snapshot` to create it.